modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/delegate/OperationModelLoaderDelegate.java

```diff
@@ -141,4 +177,81 @@ private void declareErrorModels(OperationDeclarer operation, OperationModelParse
+                                                 Optional<OutputResolverModelParser> outputResolverModelParser,
+                                                 List<InputResolverModelParser> inputResolverModelParsers,
+                                                 Optional<KeyIdResolverModelParser> keyIdResolverModelParser) {
+    if (keyIdResolverModelParser.isPresent()) {
+      String parameterName = keyIdResolverModelParser.get().getParameterName();
+      MetadataType metadataType = keyIdResolverModelParser.get().getMetadataType();
+      String categoryName = JavaModelLoaderUtils.getCategoryName(keyIdResolverModelParser.orElse(null), inputResolverModelParsers,
+                                                                 outputResolverModelParser.orElse(null));
+
+      operation.withModelProperty(new MetadataKeyIdModelProperty(metadataType, parameterName, categoryName));
+    }
+  }
+
+  private Optional<KeyIdResolverModelParser> getKeyIdResolverModelParser(OperationModelParser operationModelParser,
+                                                                         OutputResolverModelParser outputResolverModelParser,
+                                                                         AttributesResolverModelParser attributesResolverModelParser,
+                                                                         List<InputResolverModelParser> inputResolverModelParsers) {
+
+    Optional<KeyIdResolverModelParser> keyIdResolverModelParser = empty();
+    if (outputResolverModelParser != null || !inputResolverModelParsers.isEmpty()) {
+      String categoryName = getCategoryName(outputResolverModelParser, attributesResolverModelParser, inputResolverModelParsers);
+
+      keyIdResolverModelParser = operationModelParser.getParameterGroupModelParsers().stream()
+          .map(parameterGroupModelParser -> parameterGroupModelParser.getKeyIdResolverModelParser(categoryName))
+          .filter(Optional::isPresent)
+          .map(Optional::get)
+          .findFirst();
+
+      if (!keyIdResolverModelParser.isPresent()) {
+        keyIdResolverModelParser = operationModelParser.getParameterGroupModelParsers().stream()
+            .map(ParameterGroupModelParser::getParameterParsers)
+            .flatMap(List::stream)
+            .collect(toList())
+            .stream()
+            .map(parameterModelParser -> parameterModelParser.getKeyIdResolverModelParser(categoryName))
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .findFirst();
+      }
+
+      if (keyIdResolverModelParser.isPresent() && !keyIdResolverModelParser.get().hasKeyIdResolver()) {
+        Optional<KeyIdResolverModelParser> enclosingKeyIdResolverModelParser = operationModelParser.getKeyIdResolverModelParser();
+        if (enclosingKeyIdResolverModelParser.isPresent()) {
+          keyIdResolverModelParser = of(new JavaKeyIdResolverModelParser(
```

Comment:
OperationModelLoaderDelagate should be agnostic to java, it shouldn't instance a JavaKeyIdResolverModelParser.